src/main/java/org/jenkinsci/plugins/vsphere/RunOnceCloudRetentionStrategy.java

```diff
@@ -176,4 +178,12 @@ public String getDisplayName() {
             return "vSphere Run-Once Retention Strategy";
         }
     }
+
+    @Extension
+    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
+        @Override
+        public boolean filter(@CheckForNull Object context, @NonNull Descriptor descriptor) {
+            return !(descriptor instanceof DescriptorImpl);

```

Comment:
I don't see how this can be correct for all situations in which it'll be called.
My (possibly mistaken) understanding is that, when a user is manually creating a new (slave) Node, they can choose to create a vSphere node (with the ability to start/stop/revert-to-snapshot etc) and, in that `context` it'll be necessary to be able to select this strategy.
If `context` isn't a vSphere slave _then_ we don't allow this Descriptor through the filter, but I think we have to allow it through for vSphere slaves.

Reply:
I'll look into the correct context.